scorekeeper/src/App.test.js

```diff
@@ -0,0 +1,153 @@
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/strike/i);
+    const currentStrikes = getAllByTestId('strikes')[0];
+    fireEvent.click(button);
+    expect(currentStrikes).toHaveTextContent(/Strikes: 1/i)
+    cleanup();
+  })
+
+  // after 3 strikes, everything resets to 0
+  it('should reset all values to 0 after 3 strikes', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/strike/i);
+    const currentStrikes = getAllByTestId('strikes')[0];
+    const currentBalls = getAllByTestId('balls')[0];
+    const currentFouls = getAllByTestId('fouls')[0];
+    const currentHits = getAllByTestId('hits')[0];
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    expect(currentStrikes).toHaveTextContent(/Strikes: 0/i)
+    expect(currentBalls).toHaveTextContent(/balls: 0/i)
+    expect(currentFouls).toHaveTextContent(/fouls: 0/i)
+    expect(currentHits).toHaveTextContent(/hits: 0/i)
+    cleanup();
+  })
+})
+  
+  describe('Ball button', () => {
+// test ball button
+it('should increment balls by 1 when ball button is clicked', () => {
+  const { getByText, getAllByTestId } = render(<App />);
+  const button = getByText(/ball/i);
+  const currentBalls = getAllByTestId('balls')[0];
+  fireEvent.click(button);
+  expect(currentBalls).toHaveTextContent(/Balls: 1/i)
+  cleanup();
+})
+
+// after 4 balls, everything resets to 0
+it('should reset all values to 0 after 4 balls', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/ball/i);
+    const currentStrikes = getAllByTestId('strikes')[0];
+    const currentBalls = getAllByTestId('balls')[0];
+    const currentFouls = getAllByTestId('fouls')[0];
+    const currentHits = getAllByTestId('hits')[0];
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    expect(currentStrikes).toHaveTextContent(/Strikes: 0/i)
+    expect(currentBalls).toHaveTextContent(/balls: 0/i)
+    expect(currentFouls).toHaveTextContent(/fouls: 0/i)
+    expect(currentHits).toHaveTextContent(/hits: 0/i)
+    cleanup();
+  })
+})
+
+
+describe('Foul button', () => {
+  // test foul button
+  it('should increment fouls by 1 when foul button is clicked', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/foul/i);
+    const currentFouls = getAllByTestId('fouls')[0];
+    fireEvent.click(button);
+    expect(currentFouls).toHaveTextContent(/fouls: 1/i)
+    cleanup();
+  })
+
+  // test foul button when fouls have reached 2
+  it('fouls value should stay at 2 after more than 2 fouls', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/foul/i);
+    const currentFouls = getAllByTestId('fouls')[0];
```

Comment:
An alternative to array indexing to get the first (and I assume only) element here is array destructuring:
```javascript
const [ currentFouls ] = getAllByTestId('fouls');
```
which I typically prefer for picking off the first element of an array (but it doesn't really make a difference; it's doing the same thing). The difference it does make is visual; with destructuring, it's right where the variable name is declared, literally surrounding it, while with indexing it is at the very end and can be harder to pick up on.